src/compiler/diagnostic/tsconfig.json

```diff
@@ -0,0 +1,10 @@
+{ 
+  "extends": "../../../tsconfig-base.json",
+  "compilerOptions": {
+    "composite": true,
+    "strict": true
+  },
+  "include": [
+    "./index.ts"
+  ]
+}
```

Comment:
this is the simplest sub-project I created. It just includes a single file (`src/compiler/diagnostic/index.ts`) and doesn't depend on any other projects.

Note that if you try to import from outside of the current project into a file that is not included in it TypeScript will give you an error, requiring you do explicitly define and delineate the relationships among the subprojects. I think this is a good thing!